pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -443,6 +491,11 @@ func (r *ReconcileKubeDirectorCluster) handleNewCluster(
 			shared.EventReasonNoEvent,
 			"ignoring cluster CR with stale status UID; will retry",
 		)
+		// If ConfigMetaGenerator has been updated then ignore UID check and
+		// return
+		if cr.Spec.ConfigMetaGenerator != cr.Status.LastConfigMetaGenerator {
+			return true, nil
+		}
```

Comment:
This alarms me a little bit. Were you seeing the mismatch error message a lot?

Reply:
Yes there is mismatch error otherwise

Reply:
When does the error happen? When you edit a kdcluster to change its connections? Or when you edit some other cluster or configmap that this one is connected to?

This particular case in this function means that a cluster CR is coming in with a gen UUID in the status that is different from the value we last wrote there. This is a worrying thing because we should be the only entity allowed to write to the status. The only way it can happen in normal operation is when the read cache sends us an old version of the object, which doesn't yet reflect the changes we most recently wrote to it. If this happens we _must_ drop the object without processing it, it can cause real shenanigans otherwise.

Reply:
I guess this was coming due to extensive re-deploying in my last round of testing, I also fixed certain sequencing since the last testing and not seeing this anymore. So I will remove this. Thanks for pointing this out.